src/setup.ts

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import path from 'path'
+import * as core from '@actions/core'
+import * as exec from '@actions/exec'
+import * as sdk_utils from './sdk'
+import * as system from './system'
+import * as oidc from './oidc'
+import * as versions from './versions'
+import * as tc from '@actions/tool-cache'
+
+// TODO: Support version patterns of '2.19.x' to mean the latest 2.19 version.
```

Comment:
I believe mono_repo has a feature to generate tests using the lower bound of the package, maybe setup-dart could also have that feature?
version = auto -> find pubspec yaml (or perhaps lock) and use the lower bound.